src/transports/smart_protocol.c

```diff
@@ -751,9 +751,10 @@ static int add_push_report_sideband_pkt(git_push *push, git_pkt_data *data_pkt,
 				git_buf_put(data_pkt_buf, line, line_len);
 			error = 0;
 			goto done;
-		}
-		else if (error < 0)
+		} else if (error) {
```

Comment:
Oh, never mind this one. I've been confused by the analysis, as well, and first thought it was a false positive. I then went and checket the function and actually saw what I wanted to see in there I guess, so I misread two different locations and thought it obvious that positive values might be returned.

Will drop the patch. Thanks for double-checking.